architecture.md

```diff
@@ -0,0 +1,44 @@
+# Architecture
+
+This document describes the high-level architecture of the Gleam transpiler.
+
+## Overview
+
+On the highest level, a `gleam` binary receives Gleam source code as input
+and produces one of its available backend formats:
+
+- Erlang source code (not Erlang Abstract Format, as e.g. Elixir does)
+- Javascript source code
+
+There is currently no formal specification of Gleam source,
+apart from the programmer-oriented [Language tour](https://gleam.run/book/tour/index.html).
+
+The code transpilation is for the most part oriented around the properties
+provided by Erlang's BEAM machine, including:
```

Comment:
While it could be good to talk about this stuff I think it would be better suited to some kind of design documentation rather than architecture documentation.